File: app/src/main/java/com/example/spring/CreatePlanRequest.java
package com.example.spring;

/**
 * Request DTO for command CreatePlan.
 */
public record CreatePlanRequest(
    @jakarta.validation.constraints.NotBlank String billingInterval,
    @jakarta.validation.constraints.NotBlank String createdAt,
    @jakarta.validation.constraints.NotBlank String displayName,
    @jakarta.validation.constraints.NotBlank String planCode,
    @jakarta.validation.constraints.NotBlank String planId,
    @jakarta.validation.constraints.NotBlank String status,
    @jakarta.validation.constraints.NotBlank String tenantId
) {}

File: app-python/entitlements_app/requests/create_plan_request.py
"""Pydantic request model for command `CreatePlan`."""

from pydantic import BaseModel


class CreatePlanRequest(BaseModel):
    billingInterval: str
    createdAt: str
    displayName: str
    planCode: str
    planId: str
    status: str
    tenantId: str

    @property
    def billing_interval(self) -> str:
        return self.billingInterval

    @property
    def created_at(self) -> str:
        return self.createdAt

    @property
    def display_name(self) -> str:
        return self.displayName

    @property
    def plan_code(self) -> str:
        return self.planCode

    @property
    def plan_id(self) -> str:
        return self.planId

    @property
    def tenant_id(self) -> str:
        return self.tenantId
